core/src/replay_stage.rs

```diff
@@ -322,13 +322,20 @@ impl ReplayStage {
         }
         locktower.update_epoch(&bank);
         if let Some(ref voting_keypair) = voting_keypair {
+            let node_keypair = cluster_info.read().unwrap().keypair.clone();
+
             // Send our last few votes along with the new one
-            let vote_ix = vote_instruction::vote(vote_account_pubkey, locktower.recent_votes());
-            let vote_tx = Transaction::new_signed_instructions(
-                &[voting_keypair.as_ref()],
-                vec![vote_ix],
-                bank.last_blockhash(),
+            let vote_ix = vote_instruction::vote(
+                &node_keypair.pubkey(),
+                &vote_account,
+                &voting_keypair.pubkey(),
+                locktower.recent_votes(),
             );
+
+            let mut vote_tx = Transaction::new_unsigned_instructions(vec![vote_ix]);
+            let blockhash = bank.last_blockhash();
+            vote_tx.partial_sign(&[node_keypair.as_ref()], blockhash);
+            vote_tx.partial_sign(&[voting_keypair.as_ref()], blockhash);
```

Comment:
@garious, rust can't handle `&[node_keypair.as_ref(), voting_keypair.as_ref()]`

Reply:
That's probably only true if they are different implementations of KeypairUtil.  Your workaround of making two calls is a very good one. The alternative is to change the APIs to take trait objects, but that a malloc to box every keypair reference (which I'm okay with).

Reply:
> That's probably only true if they are different implementations of KeypairUtil.
nope.  both node_keypair and voting_keypair come from the same place...

@sagar-solana and I looked at it and found some really weird stuff going on with rust